src/BrowserView.m

```diff
@@ -207,12 +207,25 @@ -(void)closeAllTabs
  */
 -(void)closeTabItemView:(NSView *)tabItemView
 {
-	NSTabViewItem *tabViewItem = [self.tabBarControl tabViewItemWithIdentifier:tabItemView];
-
-	[self tabView:self.tabBarControl.tabView willCloseTabViewItem:tabViewItem];
-	//close tab to be closed
-	[self.tabBarControl removeTabViewItem:tabViewItem];
-	[self tabView:self.tabBarControl.tabView didCloseTabViewItem:tabViewItem];
+	MMTabBarView *tabBar = self.tabBarControl;
+	NSTabView *tabView = tabBar.tabView;
+	NSTabViewItem *tabViewItem = [tabBar tabViewItemWithIdentifier:tabItemView];
+
+    if (([tabBar delegate]) && ([[tabBar delegate] respondsToSelector:@selector(tabView:shouldCloseTabViewItem:)])) {
```

Comment:
I was tempted by a quick fix similar to yours, then I hesitated : should I remove the call to `didCloseTabViewItem:` or implement it ?
The former seemed simpler, the latter seemed much safer in case a future developer would rely on what is described in the MMTabBarView header ! He would probably fall in the same trap we fell… So I chose the safe way and decided to reproduce what the Demo app does.

Thinking further, I now believe that even that is not very good… What we did is just an adaptation of MMTabBarView's __inner code__ in Vienna's code. Such a workaround goes completely against the aim of good object programming… The clean solution would be having a closeTab: method directly exposed by MMTabBarView.

Reply:
You are right, the tab bar should be controllable only through its own interface and not through the NSTabView methods etc.. I found this to be very annoying when I tried to decouple the tab view and the browser, by only using the`tabbarcontroller` reference. Sometimes I had to access `tabbarcontroller.tabview` to achieve the functionality needed.

Reply:
Another solution would be modifying MMTabBarView's `removeTabViewItem:` to include the necessary stuff.